static/js/settings_emoji.js

```diff
@@ -260,15 +260,26 @@ export function set_up() {
             if (is_default_emoji(emoji.name.replace(/ /g, "_"))) {
                 // spaces are replaced with `_` to simulate how the emoji
                 // name will actually be stored in the backend
-                const html_body = emoji_settings_warning_modal({
-                    emoji_name: emoji.name,
-                });
 
-                confirm_dialog.launch({
-                    html_heading: $t_html({defaultMessage: "Override default emoji?"}),
-                    html_body,
-                    on_click: submit_custom_emoji_request,
-                });
+                if (page_params.is_admin) {
+                    const html_body = emoji_settings_warning_modal({
+                        emoji_name: emoji.name,
+                    });
+
+                    confirm_dialog.launch({
+                        html_heading: $t_html({defaultMessage: "Override default emoji?"}),
+                        html_body,
+                        on_click: submit_custom_emoji_request,
+                    });
+                } else {
+                    ui_report.client_error(
```

Comment:
I think it would be better to replace this if-else block with an early return block for `!page_params.is_admin`.